### PR TITLE
feat(csi-secret-provider-class): intialize chart

### PR DESCRIPTION
**Pull request by hairmare**

### CODE REVIEW

File: charts/csi-secret-provider-class/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+name: csi-secret-provider-class
```
Comment (eyenx) [inline charts/csi-secret-provider-class/Chart.yaml:2]:
```suggestion
name: csi-secret-provider-class
home: https://github.com/adfinis-sygroup/helm-charts/tree/master/charts/csi-secret-provider-class
```

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/csi-secret-provider-class/values.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# provider -- specify secret provider
+provider: ~
```
Comment (eyenx) [inline charts/csi-secret-provider-class/values.yaml:2]: why exactly `~` and not `""` ?

File: charts/csi-secret-provider-class/templates/secretproviderclass.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+{{ template "common.metadata" $ }}
+spec:
+  provider: {{ required "Specify a valid provider." .Values.provider }}
```
Comment (eyenx) [inline charts/csi-secret-provider-class/templates/secretproviderclass.yaml:5]: if default value for `provider` is `~` why set it as `required` ?

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/csi-secret-provider-class/values.yaml (hunk 2 above)
Comment (hairmare) [reply charts/csi-secret-provider-class/values.yaml:2]: `~` is `null` in YAML and it is used to indicate that this is per default falsy and needs to be set.

File: charts/csi-secret-provider-class/templates/secretproviderclass.yaml (hunk 3 above)
Comment (hairmare) [reply charts/csi-secret-provider-class/templates/secretproviderclass.yaml:5]: the falsy value of `~` triggers the required warning.

File: charts/csi-secret-provider-class/Chart.yaml (hunk 1 above)
Comment (hairmare) [reply charts/csi-secret-provider-class/Chart.yaml:2]: yepp, i pushed this together with the other things needed for the linter to pass

File: charts/csi-secret-provider-class/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+## Values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| parameterTpl | string | raw values from `parameters` | template to generate parameters |
+| parameters | object | no parameters | parameters for provider |
+| provider | string | `nil` | specify secret provider |
+| secretObjects | list | `[]` | secretObjects for for SecretProviderClass |
+| secretObjectsTpl | string | `"{{ .Values.secretObjects | toYaml }}"` |  |
```
Comment (hairmare) [inline charts/csi-secret-provider-class/README.md:30]: this doesn't escape properly :P

**Review by eyenx — APPROVED** [review]

**Comment by eyenx** [comment]
:fireworks:

**Comment by eyenx** [comment]
:fireworks:

**Pull request merged**